docs/src/components/home/sections/PrimitiveSection.tsx

```diff
@@ -0,0 +1,245 @@
+import { useRef, useState } from 'react';
+import classNames from 'classnames';
+import {
+  MdFormatAlignCenter,
+  MdFormatAlignJustify,
+  MdFormatAlignLeft,
+  MdFormatAlignRight,
+  MdOutlineWidgets,
+} from 'react-icons/md';
+import {
+  Heading,
+  Text,
+  View,
+  Card,
+  Flex,
+  CheckboxField,
+  RadioGroupField,
+  Radio,
+  SliderField,
+  SearchField,
+  SwitchField,
+  Tabs,
+  TabItem,
+  Badge,
+  ToggleButtonGroup,
+  ToggleButton,
+  Button,
+  Pagination,
+} from '@aws-amplify/ui-react';
+import { HomeCTA } from '@/components/home/HomeCTA';
+import { useIntersectionObserver } from '@/components/useIntersection';
+import { upperFirst } from 'lodash';
+
+const Center = ({ children }) => (
+  <Flex justifyContent="center" alignItems="center" width="100%" height="100%">
+    {children}
+  </Flex>
+);
+
+const Preview = () => {
+  const [exclusiveValue, setExclusiveValue] = useState('align-left');
+  const [radioValue, setRadioValue] = useState('css');
+  const [currentPageIndex, setCurrentPageIndex] = useState(4);
+  const totalPages = 99;
+
+  const handleOnChange = (newPageIndex, prevPageIndex) => {
+    setCurrentPageIndex(newPageIndex);
+  };
+
+  const direction = ['column', 'column', 'row'];
+
+  return (
+    <Flex direction="column" width="100%">
+      <Flex direction={direction}>
+        <Card variation="elevated">
+          <Center>
+            <SwitchField label="switch" labelPosition="end" />
+          </Center>
+        </Card>
+        <Card variation="elevated" flex="1" display={['none', 'block']}>
+          <Center>
+            <Pagination
```

Comment:
Clicking the "next" and "prev" buttons for the Pagination component does not change the currently selected page. Can we make this a controlled component so that the next/prev button functionality works?